cpp/oneapi/dal/algo/linear_kernel/backend/gpu/compute_kernel_dense_dpc.cpp

```diff
@@ -14,20 +14,84 @@
 * limitations under the License.
 *******************************************************************************/
 
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+
+#define DAAL_SYCL_INTERFACE
+#include <daal/src/algorithms/kernel_function/oneapi/kernel_function_linear_dense_default_kernel_oneapi.h>
+
 #include "oneapi/dal/algo/linear_kernel/backend/gpu/compute_kernel.hpp"
+#include "oneapi/dal/backend/interop/common.hpp"
 
 namespace oneapi::dal::linear_kernel::backend {
 
+using dal::backend::context_gpu;
+
+namespace daal_linear_kernel = daal::algorithms::kernel_function::linear;
+namespace interop            = dal::backend::interop;
+
+template <typename Float>
+using daal_linear_kernel_t =
+    daal_linear_kernel::internal::KernelImplLinearOneAPI<daal_linear_kernel::defaultDense, Float>;
+
+template <typename Float>
+static compute_result call_daal_kernel(const context_gpu& ctx,
+                                       const descriptor_base& desc,
+                                       const table& x,
+                                       const table& y) {
+    const int64_t row_count_x  = x.get_row_count();
+    const int64_t row_count_y  = y.get_row_count();
+    const int64_t column_count = x.get_column_count();
+
+    auto queue = ctx.get_queue();
+    auto arr_x = row_accessor<const Float>{ x }.pull(queue);
+    auto arr_y = row_accessor<const Float>{ y }.pull(queue);
+
+    auto arr_values = array<Float>::empty(queue, row_count_x * row_count_y);
+
+    daal::services::SyclExecutionContext daal_ctx(queue);
+    daal::services::Environment::getInstance()->setDefaultExecutionContext(daal_ctx);
```

Comment:
Create `execution_context_guard` in `backend::interop` as we discussed

Reply:
Applied